### PR TITLE
Use httpx over aiohttp

### PR DESCRIPTION
**Pull request by gaborbernat**
Resolves https://github.com/devpi/devpi/issues/998

### CODE REVIEW

**Comment by fschulze** [comment]
I tried the patch with some fixes in https://github.com/fschulze/devpi/tree/httpx and currently ``test_auth_mirror_url_no_hash`` and ``test_auth_mirror_url_with_hash`` fail, because httpx exposes the authentication info in its log output. We either need to block that from our side, or need to get a fix into httpx to remove at least the password from the output.

**Comment by fschulze** [comment]
@zanieb thanks for the offer. Currently httpx logs the full URL including username and password of each request. It is hard to make sure this doesn't leak out onto the disk and outside services. Such leaks are often the cause for security breaks. It would be best if the username and password would be obfuscated or if it wouldn't log at all. We could configure our logging to exclude it, but if anything changes in httpx, it might come back.

**Comment by zanieb** [comment]
@fschulze I took a look at the CI logs and they're pretty hard to parse :) it looks like everything fails when your secret type is logged because it's cast to a string and that throws a warning? Could you create a simpler reproduction so we can consider the best way to elide the secrets? I agree it's less than ideal to log them.

**Comment by fschulze** [comment]
@zanieb The logging in httpx happens in two places:

https://github.com/encode/httpx/blob/90d71e63e017fb5535a97c3c880d1373ae8918f4/httpx/_client.py#L1013-L1020

https://github.com/encode/httpx/blob/90d71e63e017fb5535a97c3c880d1373ae8918f4/httpx/_client.py#L1729-L1736

Any request which contains a user name and password will log that. In devpi we test if the password shows up in any log output.

You even seem to have explicit tests for the logging: ``test_logging_redirect_chain`` and ``test_logging_request``
As well as documentation:
https://github.com/encode/httpx/blob/90d71e63e017fb5535a97c3c880d1373ae8918f4/docs/logging.md?plain=1#L35

**Comment by fschulze** [comment]
BTW, it is best do obfuscate the username as well as the password. I have seen cases where the username is a token and the password is empty.

**Comment by zanieb** [comment]
Thanks! I'm actually asking for a more concrete example of the request you're sending. I'm just guessing the username and password are part of the URL?

**Comment by fschulze** [comment]
@zanieb yes ... which just lead to a thought: I could just add an ``Authorization`` header to the request instead of passing the username and password in the URL, right?

**Comment by zanieb** [comment]
That seems like a much better place for it if supported by the server! I do not think headers are logged.

**Comment by fschulze** [comment]
@zanieb Thanks for triggering that insight, it seems to work.
